nightfall-optimist/src/services/block-assembler.mjs

```diff
@@ -117,49 +126,56 @@ export async function conditionalMakeBlock(proposer) {
         });
 
         // propose this block to the Shield contract here
-        const unsignedProposeBlockTransaction = await (
-          await waitForContract(STATE_CONTRACT_NAME)
-        ).methods
+        const txDataToSign = await stateContractInstance.methods
           .proposeBlock(
             Block.buildSolidityStruct(block),
             transactions.map(t => Transaction.buildSolidityStruct(t)),
           )
           .encodeABI();
+        const { address } = web3.eth.accounts.privateKeyToAccount(ethPrivateKey);
+        const blockStake = await stateContractInstance.methods.getBlockStake().call();
+        const tx = {
+          from: address,
+          to: stateContractAddress,
+          data: txDataToSign,
+          value: blockStake,
+          gas: 8000000,
```

Comment:
Fix this constant. There should be a gas estimation and gas price as we have always had

Reply:
Addressed in #1262 
FYI I closed the previous PR for gas estimation and applied the changes here: put gas functions in common-files, preserving `gasPrice`.